### PR TITLE
Ap starting on UI

### PR DESCRIPTION
**Pull request by APatel-AI**
- Added a navbar
- Updated dropdown menu links
- Added notice messages
- Created user authentication conditions.

### CODE REVIEW

**Review by dylansarikas — APPROVED** [review]
Looks good!!  Just remember to delete the redundant alerts in the next part just as Raghu mentioned

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,4 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,11 +6,30 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+      <!--Bootsrap CSS Link-->
+  <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:10]: Nice job adding your CDN assets to a partial! This helps keep the layout file clean :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,11 +6,30 @@
...
     <%= javascript_importmap_tags %>
+
+
+
   </head>
 
   <body>
-    <%= yield %>
+
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:21]: same as above

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,30 @@
...
+    
+    <div class="container">
+      <% if notice.present? %>
+        <%= render partial: "shared/flash", locals: { message: notice, css_class: "success" } %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:25]: same as above

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+       <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <a class="dropdown-item" href="/photos">Photos</a>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:14]: remember we can start using helper methods here like link_to :)

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -10,6 +11,21 @@ task({ :sample_data => :environment }) do
     User.destroy_all
   end
 
+
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:20]: Great job on sample data!

**Review by JoseMDO — APPROVED** [review]
Everything looks good!

File: app/views/shared/_navbar.html.erb (hunk 5 above)
Comment (Cindellai) [reply app/views/shared/_navbar.html.erb:14]: Great work on updating the dropdown menu links.

**Comment by Cindellai** [comment]
@APatel-AI Great job on implementing the navbar and user authentication conditions! The dropdown menu links are clear and user-friendly.

**Review by Cindellai — COMMENTED** [review]
Great Job!

**Pull request merged**